spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/tracing/MicrometerTracingAutoConfiguration.java

```diff
@@ -70,20 +69,20 @@ public DefaultTracingObservationHandler defaultTracingObservationHandler(Tracer
 
 	@Bean
 	@ConditionalOnMissingBean
-	@ConditionalOnBean({ Tracer.class, HttpServerHandler.class })
-	@Order(HTTP_SERVER_TRACING_OBSERVATION_HANDLER_ORDER)
-	public HttpServerTracingObservationHandler httpServerTracingObservationHandler(Tracer tracer,
-			HttpServerHandler httpServerHandler) {
-		return new HttpServerTracingObservationHandler(tracer, httpServerHandler);
+	@ConditionalOnBean({ Tracer.class, Propagator.class })
+	@Order(RECEIVER_TRACING_OBSERVATION_HANDLER_ORDER)
+	public PropagatingSenderTracingObservationHandler<?> propagatingSenderTracingObservationHandler(Tracer tracer,
+			Propagator propagator) {
+		return new PropagatingSenderTracingObservationHandler<>(tracer, propagator);
 	}
 
 	@Bean
 	@ConditionalOnMissingBean
-	@ConditionalOnBean({ Tracer.class, HttpClientHandler.class })
-	@Order(HTTP_CLIENT_TRACING_OBSERVATION_HANDLER_ORDER)
-	public HttpClientTracingObservationHandler httpClientTracingObservationHandler(Tracer tracer,
-			HttpClientHandler httpClientHandler) {
-		return new HttpClientTracingObservationHandler(tracer, httpClientHandler);
+	@ConditionalOnBean({ Tracer.class, Propagator.class })
```

Comment:
This means that both receiver and sender handlers will be created if there is a propagator.